src/main/java/nextstep/subway/line/LineService.java

```diff
@@ -0,0 +1,104 @@
+                     StationService stationService, LineSectionRepository lineSectionRepository) {
+    this.lineRepository = lineRepository;
+    this.stationRepository = stationRepository;
+    this.stationService = stationService;
+    this.lineSectionRepository = lineSectionRepository;
+  }
+
+  @Transactional
+  public Line createLine(LineCreateRequest lineCreateRequest) {
+    StationsAtLine stationsAtLine = getStationsAtLine(lineCreateRequest.getUpStationId(), lineCreateRequest.getDownStationId());
+    LineSection lineSection = new LineSection(stationsAtLine.getUpStation(), stationsAtLine.getDownStation(), lineCreateRequest.getDistance());
+    return lineRepository.save(new Line());
+  }
+
+  public List<LineResponse> getLines() {
+    return lineRepository.findAll().stream()
+            .map(LineResponse::from)
+            .collect(Collectors.toList());
+  }
+
+  public LineResponse getLine(Long id) {
+    return LineResponse.from(lineRepository.findById(id).orElseThrow(() -> new SubwayException(NOT_FOUND)));
+  }
+
+  @Transactional
+  public LineResponse modifyLine(Long id, LineModifyRequest lineModifyRequest) {
+    Line line = getLineById(id);
+    line.updateName(lineModifyRequest.getName());
+    line.updateColor(lineModifyRequest.getColor());
+    return LineResponse.from(line);
+  }
+
+  @Transactional
+  public void deleteLine(Long id){
+    lineRepository.deleteById(id);
+  }
+
+  private StationsAtLine getStationsAtLine(Long upStationId, Long downStationId) {
+    Station upStation = stationRepository.findById(upStationId).orElseThrow(()-> new SubwayException(NOT_FOUND));
+    Station downStation = stationRepository.findById(downStationId).orElseThrow(()-> new SubwayException(NOT_FOUND));
+
+    return new StationsAtLine(upStation, downStation);
+  }
+
+  private Line getLineById(Long id) {
+    return lineRepository.findById(id).orElseThrow(()-> new SubwayException(NOT_FOUND));
+  }
+
+  @Transactional
+  public void appendLineSection(Long id, LineSectionAppendRequest lineSectionAppendRequest) {
+    Line line = getLineById(id);
+    Station upStation = stationService.findById(lineSectionAppendRequest.getUpStationId());
+    Station downStation = stationService.findById(lineSectionAppendRequest.getDownStationId());
+
+    LineSection lineSection = new LineSection(upStation, downStation, lineSectionAppendRequest.getDistance());
+    line.appendSection(lineSection);
+    lineSectionRepository.save(lineSection);
+  }
+
+  @Transactional
+  public void deleteLineSection(Long lineId, Long stationId) {
+    List<LineSection> lineSection = lineSectionRepository.findAllByLineId(lineId);
+
+    LineSection lastLineSection = lineSection.stream().max((a, b) -> (int) (a.getIndex() - b.getIndex()))
+      .orElseThrow(() -> new SubwayException(NOT_FOUND));
+
+    if (lastLineSection.getDownStation().getId() != stationId || lineSection.size() <= 1) {
```

Comment:
```suggestion
    if (!lastLineSection.getDownStation().getId().eqauls(stationId) || lineSection.size() <= 1) {
```

Wrapper type 은 `==` `!=` 보다는 equals 로 비교해주세요 
[참고](https://dami97.tistory.com/24)